src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java

```diff
@@ -332,7 +344,40 @@ public static FeedSource checkFeedSourcePermissions(Request req, FeedSource feed
             // Throw halt if user not authorized.
             logMessageAndHalt(req, 403, "User not authorized to perform action on feed source");
         }
+
+
         // If we make it here, user has permission and the requested feed source is valid.
+        // This final step removes labels the user can't view
+        return cleanFeedSourceLabels(feedSource, userProfile);
+    }
+
+    /** Determines whether a change to a feed source is significant enough that it warrants sending a notification
+     *
+     * @param formerFeedSource  A feed source object, without new changes
+     * @param updatedFeedSource A feed source object, with new changes
+     * @return                  A boolean value indicating if the updated feed source is changed enough to warrant sending a notification.
+     */
+    private static boolean shouldNotifyUsersOnFeedUpdated(FeedSource formerFeedSource, FeedSource updatedFeedSource) {
+        return
+                // If only labels have changed, don't send out an email
+                formerFeedSource.equalsExceptLabels(updatedFeedSource);
+    }
+
+    /**
+     * Removes labels from a feed that a user is not allowed to view. Returns cleaned feed source
+     * @param feedSource    The feed source to clean
+     * @param userProfile   The user object to check permissions with
+     * @return              A feed source containing only labels the user is allowed to see
+     */
+    private static FeedSource cleanFeedSourceLabels(FeedSource feedSource, Auth0UserProfile userProfile) {
+        // Remove labels user is not allowed to see if user is not admin
+        boolean isAdmin = userProfile.canAdministerProject(feedSource.id, feedSource.organizationId());
+        if (!isAdmin) {
+            feedSource.labels = Persistence.labels.getByIds(feedSource.labels).stream()
+                    .filter(label -> !label.adminOnly)
```

Comment:
I don't have very strong feelings about this since it won't be very many records, but it seems like filtering out the `adminOnly` labels could be something done as part of the database query to ensure that the minimum required data is transferred over the network from the db to this application.

Reply:
Have refactored this in both fa4ba57 and then further in 9ac824a. I think it's more efficient, but would appreciate your double-check

Reply:
Looks much better. Thanks!